File: Album_Store_BackEnd/AlbumStore/AlbumStore.Domain/Entities/ProductVersion.cs

namespace AlbumStore.Domain.Entities;
public class ProductVersion
{
    public Guid Id { get; set; }
    public string Version { get; set; }
    public string Description { get; set; }
    public DateTime ReleaseDate { get; set; }
    public string? ImageUrl { get; set; }
    public double Price { get; set; }
    public virtual Guid ProductId { get; set; }
    public virtual Product Product { get; set; }

}

File: AlbumStoreUI/app/src/main/java/com/albumstore/todo/data/product/ProductVersion.kt
package com.albumstore.todo.data.product

import org.json.JSONObject

data class ProductVersion(
    val id: String,
    val version: String,
    val description: String,
    val imageUrl: String,
    val price: Double,
    val productId: String
) {
    fun toJson(): String {
        // Convert to JSON
        val json = JSONObject()
        json.put("id", id)
        json.put("version", version)
        json.put("description", description)
        json.put("imageUrl", imageUrl)
        json.put("price", price)
        json.put("productId", productId)
        return json.toString()
    }
}
